packages/liveblocks-node/src/PermissionSet.ts

```diff
@@ -0,0 +1,146 @@
+// As defined in the source of truth in ApiScope in
+// https://github.com/liveblocks/liveblocks-cloudflare/blob/main/src/security.ts
+const ALL_PERMISSIONS = Object.freeze([
+  "room:write",
+  "room:read",
+  "room:presence:write",
+  "comments:write",
+  "comments:read",
+] as const);
+
+export type Permission = (typeof ALL_PERMISSIONS)[number];
+
+function isPermission(value: string): value is Permission {
+  return (ALL_PERMISSIONS as readonly unknown[]).includes(value);
+}
+
+const MAX_PERMS_PER_SET = 10;
+
+/**
+ * Assign this to a room or room pattern if you want to grant the user
+ * read-only permissions for this room.
+ */
+export const READ_ACCESS: readonly Permission[] = Object.freeze([
+  "room:read",
+  // "room:presence:write",  // XXX Should this be included in the READ_ACCESS set?
+  "comments:read",
+]);
+
+/**
+ * Assign this to a room or room pattern if you want to grant the user all
+ * permissions for this room.
+ */
+export const FULL_ACCESS: readonly Permission[] = Object.freeze([
+  "room:write",
+  "comments:write",
+
+  // XXX Are these implied automatically if you have the two write permissions above? If so, it would be nice to keep the JWT payload as small as possible.
+  // "room:read",
+  // "room:presence:write",
+  // "comments:read",
+]);
+
+const roomPatternRegex = /^[^*]{1,50}[*]?$/;
+
+/**
+ * Class to help you construct the exact permission set to grant a user, used
+ * when making `.authorizeUser()` calls.
+ *
+ * Usage:
+ *
+ *    const p = new Permissions();
+ *    p.allow(roomId, scopes)  // or...
+ *
+ * Example:
+ *
+ *    p.allow('my-room', FULL_ACCESS)  // Grant read + write access to 'my-room'
+ *    p.allow('my-room', READ_ACCESS)  // Grant read-only access to 'my-room'
+ *
+ * Rooms can be specified with a prefix match, if the name ends in an asterisk.
+ * In that case, access is granted to all rooms that start with that pattern:
+ *
+ *    p.allow('abc:*', FULL_ACCESS)    // Grant read + write access to all rooms that start with "abc:"
+ *
+ * You can define at most 10 room IDs or patterns in a single token, otherwise
+ * the token would become too large and unwieldy.
+ *
+ * All permissions granted are additive. You cannot "remove" permissions once
+ * you grant them. For example:
+ *
+ *    p
+ *      .allow('abc:*',   FULL_ACCESS)
+ *      .allow('abc:123', READ_ACCESS)
+ *
+ * Here, room `abc:123` would have full access. The second .allow() call only
+ * _adds_ read permissions, but that has no effect since full access
+ * permissions were already added to the set.
+ */
+export class PermissionSet {
```

Comment:
Also, I'm not particularly happy with the name `PermissionSet` yet. I previously had `Permissions`, but I wanted to avoid a name clash with the built-in [Permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API). Perhaps we should just use `Permissions` anyway and not worry about it too much?

Reply:
What about `LiveblocksPermission`?

Reply:
Could work. Would you prefer:

```typescript
import { Liveblocks, LiveblocksPermissions } from …

const liveblocks = new Liveblocks()

// Inside handler
const permissions = new LiveblocksPermissions()
``` 

or maybe do this pattern?

```typescript
import { Liveblocks } from …

const liveblocks = new Liveblocks()

// Inside handler
const permissions = new liveblocks.Permissions()
``` 

The second one seems a bit cleaner, not? Even though getting the class from the Liveblocks instance may not be too common? I don’t hate it.